### PR TITLE
Updated asset-types. 

### PR DESCRIPTION
**Pull request by micronAMZN**
Moved new asset-types topic up a level to /assets. Added product assets to table. Added more asset types. Improved descriptions. Removed old asset-file-types topic. This table is not complete. We will continue to update as new asset types are discovered.

Signed-off-by: Mike Cronin <mikecro@amazon.com>

### Submission Checklist:

* [ ] **Descriptive active voice** - Do descriptive sentences have a clear *subject* and *action verb*?
* [ ] **Answer the question at hand** - Does the documentation answer a *what*, *why*, *how*, or *where* type of question?
* [ ] **Consistency** - Does the content consistently follow the [style guide](https://o3de.org/docs/contributing/to-docs/style-guide/)?
* [ ] **Help the user** - Does the documentation show the user something *meaningful*?

### CODE REVIEW

File: content/docs/user-guide/assets/asset-types.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+| `.inputbindings` | Binds input from keyboards, gamepads, touchscreens, and other devices to runtime events. | `.inputbindings` |
+| `.postFXLayerCategories` | A list of key value pairs that define layers for Post FX assignments. | `.postFXLayerCategories` |
+| `.prefab` | A spawnable asset container that can include entities and/or other prefabs. | `.spawnable` |
+| `.stl` | A simple mesh format used in computer aided design (CAD) and 3D printing. Unlike other 3D scene formats, `.stl` assets can only contain a simple model with limited vertex attributes. | `.azmodel`</br>`.azlod`</br>`.azbuffer`</br>`.pxmesh` |
+| `.material` | Stores the properties and values, including texture assignments, that define the look of a surface for rendering. | `.azmaterial` |
+| `.physmaterial` | Stores properties and values that define the physical properties of a surface for simulation. | `.physmaterial` |
+| `.scriptcanvas` | A Script Canvas graph created with Script Canvas Editor, which is O3DE's visual scripting tool. | `.scriptcanvas_compiled`</br>`.scriptcanvas_fn_compiled`</br>`.luac` |
```
Comment (chanmosq) [inline content/docs/user-guide/assets/asset-types.md:30]:
```suggestion
| `.scriptcanvas` | A Script Canvas graph created with **Script Canvas Editor**, which is O3DE's visual scripting tool. | `.scriptcanvas_compiled`</br>`.scriptcanvas_fn_compiled`</br>`.luac` |

```

**Review by chanmosq — APPROVED** [review]
Minor style suggestions

File: content/docs/user-guide/assets/asset-types.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+| `.gradimagesettings` | Stores properties and values that define an image gradient. | `.gradimagesettings` |
+| `.inputbindings` | Binds input from keyboards, gamepads, touchscreens, and other devices to runtime events. | `.inputbindings` |
+| `.postFXLayerCategories` | A list of key value pairs that define layers for Post FX assignments. | `.postFXLayerCategories` |
+| `.prefab` | A spawnable asset container that can include entities and/or other prefabs. | `.spawnable` |
+| `.stl` | A simple mesh format used in computer aided design (CAD) and 3D printing. Unlike other 3D scene formats, `.stl` assets can only contain a simple model with limited vertex attributes. | `.azmodel`</br>`.azlod`</br>`.azbuffer`</br>`.pxmesh` |
+| `.material` | Stores the properties and values, including texture assignments, that define the look of a surface for rendering. | `.azmaterial` |
```
Comment (willihay) [inline content/docs/user-guide/assets/asset-types.md:28]: The alphabetized order is interrupted starting with `.material` appearing after `.stl`. Can you help those of us who are bothered by this and fix the alphabetization? :P

File: content/docs/user-guide/assets/asset-types.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+| `.physmaterial` | Stores properties and values that define the physical properties of a surface for simulation. | `.physmaterial` |
+| `.scriptcanvas` | A Script Canvas graph created with **Script Canvas Editor**, which is O3DE's visual scripting tool. | `.scriptcanvas_compiled`</br>`.scriptcanvas_fn_compiled`</br>`.luac` |
+
```
Comment (willihay) [inline content/docs/user-guide/assets/asset-types.md:31]: Blank line breaks the table rendering.
```suggestion
```

**Review by willihay — CHANGES_REQUESTED** [review]

**Review by willihay — APPROVED** [review]
It's still out of alphabetical order towards the end of the table, but I don't feel like it's worth pressing for another round of changes.

**Comment by willihay** [comment]
Yay! Happy now. :-)

**Pull request merged**